internalsdk/pro/pro.go

```diff
@@ -0,0 +1,252 @@
+		"email": email,
+	}, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// PaymentRedirect returns a checkout/redirect URL to be used to complete a Lantern Pro purchase with a payment provider
+func (c *proClient) PaymentRedirect(ctx context.Context, req *protos.PaymentRedirectRequest) (*PaymentRedirectResponse, error) {
+	var resp PaymentRedirectResponse
+	uc := userConfig(c.settings)
+	req.Locale = uc.GetLanguage()
+	b, _ := protojson.Marshal(req)
+	params := make(map[string]interface{})
+	json.Unmarshal(b, &params)
+	err := c.webclient.GetJSON(ctx, "/payment-redirect", params, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// PaymentMethods returns a list of plans along with payment providers and available payment methods
+func (c *proClient) PaymentMethods(ctx context.Context) (*PaymentMethodsResponse, error) {
+	var resp PaymentMethodsResponse
+	err := c.webclient.GetJSON(ctx, "/plans-v3", c.defaultParams(), &resp)
+	if err != nil {
+		return nil, err
+	}
+	b, _ := json.Marshal(resp)
+	log.Debugf("PaymentMethods response is %v", string(b))
+	return &resp, nil
+}
+
+// Plans is used to hit the legacy /plans endpoint. Deprecated.
+func (c *proClient) Plans(ctx context.Context) (*PlansResponse, error) {
+	var resp PlansResponse
+	err := c.webclient.GetJSON(ctx, "/plans", c.defaultParams(), &resp)
+	if err != nil {
+		return nil, err
+	}
+	for i, plan := range resp.Plans {
+		parts := strings.Split(plan.Id, "-")
+		if len(parts) != 3 {
+			continue
+		}
+		cur := parts[1]
+		if currency, ok := accounting.LocaleInfo[strings.ToUpper(cur)]; ok {
+			if oneMonthCost, ok2 := plan.ExpectedMonthlyPrice[strings.ToLower(cur)]; ok2 {
+				ac := accounting.Accounting{Symbol: currency.ComSymbol, Precision: 2}
+				amount := decimal.NewFromInt(oneMonthCost).Div(decimal.NewFromInt(100))
+				resp.Plans[i].OneMonthCost = ac.FormatMoneyDecimal(amount)
+			}
+		}
+	}
+	return &resp, nil
+}
+
+// UserCreate creates a new user
+func (c *proClient) UserCreate(ctx context.Context) (*UserDataResponse, error) {
+	var resp UserDataResponse
+	err := c.webclient.PostFormReadingJSON(ctx, "/user-create", nil, &resp)
+	if err != nil {
+		return nil, errors.New("error fetching user data: %v", err)
+	}
+	return &resp, nil
+}
+
+// UserData returns data associated with a user
+func (c *proClient) UserData(ctx context.Context) (*UserDataResponse, error) {
+	var resp UserDataResponse
+	err := c.webclient.GetJSON(ctx, "/user-data", nil, &resp)
+	if err != nil {
+		return nil, errors.New("error fetching user data: %v", err)
+	}
+	return &resp, nil
+}
+
+// RedeemResellerCode redeems a reseller code for the given user
+func (c *proClient) RedeemResellerCode(ctx context.Context, req *protos.RedeemResellerCodeRequest) (*protos.BaseResponse, error) {
+	var resp protos.BaseResponse
+	if err := c.webclient.PostFormReadingJSON(ctx, "/link-code-request", req, &resp); err != nil {
```

Comment:
Here it should be `/purchase` endpoint, I fixed it, while I was doing testing.

Reply:
Oh, I didn't even notice that. Thanks for fixing it!